src/python/dxpy/dx_extract_utils/InputsValidator.py

```diff
@@ -171,103 +179,95 @@ def interpret_mutually_exclusive_group(self, check):
-#             raise ResolutionError(
-#                 'Unable to resolve "'
-#                 + self.path
-#                 + '" to a data object or folder name in a project'
-#             )
-#         elif self.project != self.entity_result["describe"]["project"]:
-#             raise ResolutionError(
-#                 'Unable to resolve "'
-#                 + self.path
-#                 + "\" to a data object or folder name in '"
-#                 + self.project
-#                 + "'"
-#             )
-
-#     def assure_record_type(self):
-#         # resolving non record/cohort and permission issues
-#         if self.entity_result["describe"]["class"] != "record":
-#             err_exit(
-#                 "%s : Invalid path. The path must point to a record type of cohort or dataset"
-#                 % self.entity_result["describe"]["class"]
-#             )
-
-#     def resolve_permission(self):
-#         try:
-#             self.http_request = dxpy.DXHTTPRequest(
-#                 "/" + self.entity_result["id"] + "/visualize",
-#                 {"project": self.project, "cohortBrowser": False},
-#             )
-#         except PermissionDenied:
-#             err_exit(
-#                 "Insufficient permissions", expected_exceptions=(PermissionDenied,)
-#             )
-#         except (InvalidInput, InvalidState):
-#             err_exit(
-#                 "%s : Invalid cohort or dataset" % self.entity_result["id"],
-#                 expected_exceptions=(
-#                     InvalidInput,
-#                     InvalidState,
-#                 ),
-#             )
-#         except Exception as details:
-#             err_exit(str(details))
-
-#     def assure_dataset_version(self):
-#         # checking cohort/dataset version
-#         if self.http_request["datasetVersion"] != "3.0":
-#             err_exit(
-#                 "%s : Invalid version of cohort or dataset. Version must be 3.0"
-#                 % self.http_request["datasetVersion"]
-#             )
-
-#     def resolve_dataset_project(self):
-#         # Defining dataset project
-#         if ("Dataset" in self.http_request["recordTypes"]) or (
-#             "CohortBrowser" in self.http_request["recordTypes"]
-#         ):
-#             self.dataset_project = self.http_request["datasetRecordProject"]
-#         else:
-#             err_exit(
-#                 "%s : Invalid path. The path must point to a record type of cohort or dataset"
-#                 % self.http_request["recordTypes"]
-#             )
-
-#     def get_http_request_info(self):
-#         return self.http_request
+class PathValidator:
+    """
+    PathValidator class checks for invalid object inputs and its combination with passed arguments.
+    """
+
+    def __init__(self, parser_dict, project, entity_result, error_handler=print):
+        self.parser_dict = parser_dict
+        self.project = project
+        self.entity_result = entity_result
```

Comment:
@gabriela-reznik I would highly recommend to work with smaller, clear-scoped objects rather than an API call result directly. How about passing in (to the constructor) the describe object and call it `entity_describe`? That might set the expectations more clearly.

From the usage of the current result object, it's either `result["id"]` or `result["describe"]...`. I believe the `id` should be part of the "describe" too.